packets1/willmsgresp_test.go

```diff
@@ -1,23 +1,62 @@
 package packets1
 
 import (
+	"bytes"
 	"reflect"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+
+	pkts "github.com/energomonitor/bisquitt/packets"
 )
 
-func TestWillMsgRespStruct(t *testing.T) {
+func TestWillMsgRespConstructor(t *testing.T) {
+	assert := assert.New(t)
+
 	pkt := NewWillMsgResp(RC_ACCEPTED)
 
-	if assert.NotNil(t, pkt, "New packet should not be nil") {
-		assert.Equal(t, "*packets1.WillMsgResp", reflect.TypeOf(pkt).String(), "Type should be WillMsgResp")
-		assert.Equal(t, RC_ACCEPTED, pkt.ReturnCode, "Default ReturnCode should be RC_ACCEPTED")
+	if pkt == nil {
+		t.Fatal("New packet should not be nil")
 	}
+
+	assert.Equal("*packets1.WillMsgResp", reflect.TypeOf(pkt).String(), "Type should be WillMsgResp")
+	assert.Equal(RC_ACCEPTED, pkt.ReturnCode, "Default ReturnCode should be RC_ACCEPTED")
```

Comment:
Not sure why the word „Default“ is here. In what sense is the code the default one?

Reply:
It's a bug. In the original code, the packet was constructed in other way, the `ReturnCode` value was not explicitly set, so the "default" ("zero") value was used and checked.

I will fix it and use other value then `RC_ACCEPTED` to check it's really set (`RC_ACCEPTED` is zero value so it can be just filled-in as a default).